# Data acquisition y data wrangling 

Tenemos dos data set con las mismas columnas, es necesario saber si la información está duplicada o si son dos bases de datos distintas.

In [1]:
import pandas as pd

In [103]:
#Leemos el primer data set

df=pd.read_csv('train_strokes.csv',sep=',')
df.head()

id  gender   age  hypertension  heart_disease ever_married  \
0  30669    Male   3.0             0              0           No   
1  30468    Male  58.0             1              0          Yes   
2  16523  Female   8.0             0              0           No   
3  56543  Female  70.0             0              0          Yes   
4  46136    Male  14.0             0              0           No   

      work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0      children          Rural              95.12  18.0              NaN   
1       Private          Urban              87.96  39.2     never smoked   
2       Private          Urban             110.89  17.6              NaN   
3       Private          Rural              69.04  35.9  formerly smoked   
4  Never_worked          Rural             161.28  19.1              NaN   

   stroke  
0       0  
1       0  
2       0  
3       0  
4       0

In [104]:
df.shape

(43400, 12)

In [105]:
df['stroke'].value_counts()

0    42617
1      783
Name: stroke, dtype: int64

In [106]:
# Importamos el segundo data set

df_2=pd.read_csv('healthcare-dataset-stroke-data.csv',sep=',')

In [107]:
df_2.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [108]:
df_2.shape

(5110, 12)

Ahora los concatenamos y luego se analizarán los datos duplicados a partir del ID 

In [110]:
df_conc=pd.concat([df, df_2],join='outer', ignore_index = True)
df_conc.head()

id  gender   age  hypertension  heart_disease ever_married  \
0  30669    Male   3.0             0              0           No   
1  30468    Male  58.0             1              0          Yes   
2  16523  Female   8.0             0              0           No   
3  56543  Female  70.0             0              0          Yes   
4  46136    Male  14.0             0              0           No   

      work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0      children          Rural              95.12  18.0              NaN   
1       Private          Urban              87.96  39.2     never smoked   
2       Private          Urban             110.89  17.6              NaN   
3       Private          Rural              69.04  35.9  formerly smoked   
4  Never_worked          Rural             161.28  19.1              NaN   

   stroke  
0       0  
1       0  
2       0  
3       0  
4       0

In [118]:
df_conc.shape

(48510, 12)

Al unir ambos registros pasamos a tener un total de 48510 filas. Primero hay que chequear que los valores no sean repetidos, para eso utilizaremos el ID el cual debe ser único por paciente. 

In [119]:
dup_df=df_conc.duplicated(df_conc.columns[~df_conc.columns.isin(['id'])])

In [120]:
dup_df.value_counts()

False    44944
True      3566
dtype: int64

Hay 3566 pacientes que se repiten en ambos data set. Solo nos quedaremos con un registro de cada paciente. 

In [138]:
df_no_dup=df_conc.drop_duplicates(keep='first')
df_no_dup.head()

id  gender   age  hypertension  heart_disease ever_married  \
0  30669    Male   3.0             0              0           No   
1  30468    Male  58.0             1              0          Yes   
2  16523  Female   8.0             0              0           No   
3  56543  Female  70.0             0              0          Yes   
4  46136    Male  14.0             0              0           No   

      work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0      children          Rural              95.12  18.0              NaN   
1       Private          Urban              87.96  39.2     never smoked   
2       Private          Urban             110.89  17.6              NaN   
3       Private          Rural              69.04  35.9  formerly smoked   
4  Never_worked          Rural             161.28  19.1              NaN   

   stroke  
0       0  
1       0  
2       0  
3       0  
4       0

In [139]:
df_no_dup.shape

(44944, 12)

In [140]:
df_no_dup['stroke'].value_counts()

0    44114
1      830
Name: stroke, dtype: int64

Transformamos las columnas 

In [152]:
df_rename=df_no_dup.rename(columns={'stroke':'ACV',
                          'smoking_status':'Fumador',
                          'Residence_type':'Residencia', 
                          'work_type':'Trabajo',
                          'ever_married':'Casado',
                          'heart_disease':'Enfermedad cardiaca',
                          'hypertension':'Hipertension',
                          'age':'Edad',
                          'bmi':'IMC',
                          'gender':'Sexo',
                          'avg_glucose_level':'Glucemia'})

In [153]:
df_rename.head()

id    Sexo  Edad  Hipertension  Enfermedad cardiaca Casado  \
0  30669    Male   3.0             0                    0     No   
1  30468    Male  58.0             1                    0    Yes   
2  16523  Female   8.0             0                    0     No   
3  56543  Female  70.0             0                    0    Yes   
4  46136    Male  14.0             0                    0     No   

        Trabajo Residencia  Glucemia   IMC          Fumador  ACV  
0      children      Rural     95.12  18.0              NaN    0  
1       Private      Urban     87.96  39.2     never smoked    0  
2       Private      Urban    110.89  17.6              NaN    0  
3       Private      Rural     69.04  35.9  formerly smoked    0  
4  Never_worked      Rural    161.28  19.1              NaN    0

In [154]:
df_rename['Edad']=df_rename['Edad'].astype(int)
df_rename.head()

id    Sexo  Edad  Hipertension  Enfermedad cardiaca Casado  \
0  30669    Male     3             0                    0     No   
1  30468    Male    58             1                    0    Yes   
2  16523  Female     8             0                    0     No   
3  56543  Female    70             0                    0    Yes   
4  46136    Male    14             0                    0     No   

        Trabajo Residencia  Glucemia   IMC          Fumador  ACV  
0      children      Rural     95.12  18.0              NaN    0  
1       Private      Urban     87.96  39.2     never smoked    0  
2       Private      Urban    110.89  17.6              NaN    0  
3       Private      Rural     69.04  35.9  formerly smoked    0  
4  Never_worked      Rural    161.28  19.1              NaN    0

Hemos obtenido un data set con más datos y sin valores duplicados. Lo exportamos como un archivo .csv para los próximos trabajos

In [155]:
df_rename.to_csv('C:/Users/valei/Documents/DATA SCIENCE- CODER HOUSE/00. Trabajo final/data_stroke.csv', index = False)